### ISSUE TITLE
git-lfs asks for user/password although storage backend needs no authentication

### ISSUE THREAD

**Issue by ddanier** [github/git-lfs#243]
I'm currently writing some proof of concept storage backend for git-lfs following the API (see docs/api.md). Currently I have implemented no authentication for accessing the files. This may not be a valid production case (probably besides having no auth for some internal network), but draw attention to git-lfs always asking for a user/password. As far as HTTP works in other cases/clients (like browsers) authentication should only be needed after a HTTP 401 response.

This should be true for subsequent API call, too. As the API will return the necessary authentication headers in these cases the client should not run into some kind of new authentication anyways (see "header" API response, should include basic auth). It even may not be necessary to cache the authentication because of this.

In any case: git-lfs should only ask for authentication if really necessary.

**Comment by ddanier** [comment]
See https://github.com/ddanier/django-git-lfs ;-)

**Comment by technoweenie** [comment]
Good call :+1:

**Comment by technoweenie** [comment]
This won't make the cut for v0.5.1, but I do want to implement this SOON.

**Comment by ddanier** [comment]
This workaround seems to fix this issue in my side for now:
https://github.com/ddanier/django-git-lfs/commit/288ba9f6702a4472cbd854afab465cd8dd3a9296

It basically returns an "Authentication"-header with bogus data ("none:needed"). This way git-lfs known some authentication credentials and does skip to ask. This is of course only true if you use git-lfs over SSH to initialize this header. For normal HTTP-authentication you might just store some bogus data yourself.

**Comment by mnapoli** [comment]
+1 for this, our migration to git-lfs is blocked by this.

We want to use git-lfs to store the expected screenshots for our UI tests in [Piwik](https://github.com/piwik/piwik). However git-lfs doesn't work on Travis because it requests a password to download the files.

Note that our repository is public, so images should be downloadable by Travis (or anybody else) without any authentication.

**Comment by technoweenie** [comment]
We're going to be slipping this in with the new batch API (#415). So it will be usable on GitHub.com when Git LFS v0.6.0 ships.

**Comment by maddanio** [comment]
hey, we still have the issue on a public bitbucket repo:  https://maddanio@bitbucket.org/maddanio/opencv-ios.git
could that still be in git-lfs? happens with git-lfs/2.1.0 (GitHub; darwin amd64; go 1.8.1)

**Comment by ttaylorr** [comment]
@maddanio BitBucket could be asking for authentication even though none is needed. Please open a new issue if you want to discuss further.
